File: src/main/java/io/curiousoft/ijudi/ordermanagement/model/Basket.java
package io.curiousoft.ijudi.ordermanagement.model;

import javax.validation.constraints.NotEmpty;
import java.util.ArrayList;
import java.util.List;

public class Basket {

    private String id;
    @NotEmpty(message = "order basket is empty")
    private List<BasketItem> items = new ArrayList<>();

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public List<BasketItem> getItems() {
        return items;
    }

    public void setItems(List<BasketItem> items) {
        this.items = items;
    }
}

File: izinga-commons/src/main/kotlin/io/curiousoft/izinga/commons/model/Basket.kt
package io.curiousoft.izinga.commons.model

import javax.validation.constraints.NotEmpty

class Basket {
    var id: String? = null
    var items: @NotEmpty(message = "order basket is empty") MutableList<BasketItem> = ArrayList()
    val totalPrice: Double
        get() = items.sumOf { it.totalPrice }
    val totalDiscount: Double
        get() = items.sumOf { it.discountedPrice }
    val totalWeight: Double
        get() = items.sumOf { it.weight * it.quantity }
    val totalVolume: Double
        get() = items.sumOf { it.width * it.length * it.height * it.quantity }
    val totalArea: Double
        get() = items.sumOf { it.width * it.length * it.quantity }
}